### ISSUE TITLE
Not able to `instance link` a remote instance with a password

### ISSUE THREAD

**Issue by colinhacks** [edgedb/edgedb-cli#801]
```
$ edgedb instance link --dsn edgedb://edgedb:password@flieredb.internal:8080
[2022-08-03T03:45:13Z WARN  edgedb::version_check] Newer version of edgedb tool exists 2.0.2+a099121 (current 2.0.1+b407f6f). To upgrade run `edgedb cli upgrade`
Specify the host of the server [default: flieredb.internal]: 
> flieredb.internal
Specify the port of the server [default: 8080]: 
> 8080
Specify the database user [default: edgedb]: 
> edgedb
Specify the database name [default: edgedb]: 
> edgedb
Unknown server certificate: SHA1:c946c9e8645a5791c6d871b3b3b07b2d98d702e5. Trust? [y/N]
> y
Password for 'edgedb': 
Specify a new instance name for the remote server [default: flier]: 
> flier
Successfully linked to remote instance. To connect run:
  edgedb -I flier
$ edgedb -I flier
[2022-08-03T03:45:34Z WARN  edgedb::version_check] Newer version of edgedb tool exists 2.0.2+a099121 (current 2.0.1+b407f6f). To upgrade run `edgedb cli upgrade`
edgedb error: PasswordRequired: Password required for the specified user/host
$ edgedb -I flier --password
Password for 'edgedb': <enter password>
EdgeDB 2.0+59694c0 (repl 2.0.1+b407f6f)
Type \help for help, \quit to quit.
edgedb> 
```

No password is stored in `<root>/credentials/flier.json`. Is this intentional? If so, is there a reason we're prompting for a password as part of of the `instance link` flow?

**Comment by elprans** [comment]
Looks like a regression to me

**Comment by tailhook** [comment]
Fixed in 62369d5125d241ee054f47ed5f81725f70d60de6
